### ISSUE TITLE
Pressing CTRL-[ doesn't always leave insert mode.

### ISSUE THREAD

**Issue by TheBB** [emacs-evil/evil#570]
Originally reported by: **Adam Olsen (Bitbucket: [synic](https://bitbucket.org/synic), GitHub: [synic](https://github.com/synic))**

----------------------------------------

Using evil-20150818.1313 with spacemacs develop.

I have a video here.   Sorry the video is so large.  I was going to crop it down to the size of the editor, but screenflow won't let me do that and keep the keystrokes.

https://www.youtube.com/watch?v=WB42kaCP3x0

Basically, every once in a while, when I'm typing quickly, `CTRL-[` won't actually escape, but will act as though it's starting a key combination.

I am fine to change my `ESC` binding to something else (other than `ESC` itself) if something is known to work better.

----------------------------------------
- Bitbucket: https://bitbucket.org/lyro/evil/issue/570
